daemon/cluster/executor/container/executor.go

```diff
@@ -143,9 +143,23 @@ func (e *executor) Configure(ctx context.Context, node *api.Node) error {
 	attachments := make(map[string]string)
 
 	for _, na := range node.Attachments {
+		if na == nil || na.Network == nil {
+			// this should not happen, but we got a panic here and don't have a
+			// good idea about what the underlying data structure looks like.
+			logrus.WithField("NetworkAttachment", fmt.Sprintf("%#v", na)).
+				Warnf("skipping nil node attachment entry")
+			continue
+		}
+
 		if na.Network.Spec.Ingress {
 			ingressNA = na
 		}
+
+		if len(na.Addresses) == 0 {
+			logrus.WithField("NetworkAttachment", fmt.Sprintf("%#v", na)).
+				Warnf("skipping node attachment entry with no addresses")
+			continue
+		}
 		attachments[na.Network.ID] = na.Addresses[0]
 	}
 
```

Comment:
The code block likely needs to check for `len(ingressNA.Addresses) > 0` as well;

```go
 	if (ingressNA == nil) && (node.Attachment != nil) {
 		ingressNA = node.Attachment
 		attachments[ingressNA.Network.ID] = ingressNA.Addresses[0]
 	}
```